lisa/trace.py

```diff
@@ -352,6 +354,80 @@ def ftrace(self):
         """
         return self._ftrace
 
+
+    @classmethod
+    @contextlib.contextmanager
+    def from_target(cls, target, events=[], buffer_size=10240, filepath=None, **kwargs):
+        """
+        Context manager that can be used to collect a :class:`Trace` directly
+        from a :class:`lisa.target.Target` without needing to setup an
+        :class:`FtraceCollector`.
+
+        **Example**::
+
+            from lisa.trace import Trace
+            from lisa.target import Target
+
+            target = Target.from_default_conf()
+
+            with Trace.from_target(target, events=['sched_switch', 'sched_wakeup']) as trace:
+                target.execute('echo hello world')
+
+            trace.analysis.tasks.plot_tasks_total_residency(filepath='plot.png')
+
+
+        :param target: Target to connect to.
+        :type target: Target
+
+        :param events: ftrace events to collect and parse in the trace.
+        :type events: list(str)
+
+        :param buffer_size: Size of the ftrace ring buffer.
+        :type buffer_size: int
+
+        :param filepath: If set, the trace file will be saved at that location.
+            Otherwise, a temporary file is created and removed as soon as the
+            parsing is finished.
+        :type filepath: str or None
+
+        :Variable keyword arguments: Forwarded to :class:`Trace`.
+        """
+        ftrace_coll = FtraceCollector(target, events=events, buffer_size=buffer_size)
+        plat_info = target.plat_info
+
+        class TraceProxy(TraceBase):
+            def get_view(self, *args, **kwargs):
+                return self.base_trace.get_view(*args, **kwargs)
+
+            def __getattr__(self, attr):
+                return getattr(self.base_trace, attr)
+
+        proxy = TraceProxy()
+
+        with ftrace_coll:
+            yield proxy
```

Comment:
IIUC when we yield this it's actually "empty", but once we go through the equivalent of `__exit__()` we populate its `base_trace` which makes it work.

It seems obvious, but maybe we should add a warning in the docstring that the yielded object is only really usable *outside* of the `with` statement?

Reply:
True, or maybe I should raise an explicit exception if it used inside the `with` statement.